climada/engine/supplychain.py

```diff
@@ -0,0 +1,342 @@
+from tqdm import tqdm
+import numpy as np
+import pandas as pd
+from iso3166 import countries_by_numeric
+
+from climada import CONFIG
+from climada.util import files_handler as u_fh
+from climada.engine import Impact
+from climada.entity.exposures.base import Exposures
+
+LOGGER = logging.getLogger(__name__)
+WIOD_FILE_LINK = CONFIG.engine.supplychain.resources.wiod16.str()
+"""Link to the 2016 release of the WIOD tables."""
+
+WIOD_DIRECTORY = CONFIG.engine.supplychain.local_data.wiod.dir()
+"""Directory where WIOD tables are downloaded into."""
+
+class SupplyChain():
+    """SupplyChain class.
+
+    The SupplyChain class provides methods for loading Multi-Regional Input-Output
+    Tables (MRIOT) and computing direct, indirect and total impacts.
+
+    Attributes
+    ----------
+    mriot_data : np.array
+        The input-output table data.
+    mriot_reg_names : np.array
+        Names of regions considered in the input-output table.
+    sectors : np.array
+        Sectors considered in the input-output table.
+    total_prod : np.array
+        Countries' total production.
+    mriot_type : str
+        Type of the adopted input-output table.
+    reg_pos : dict
+        Regions' positions within the input-output table and impact arrays.
+    reg_dir_imp : list
+        Regions undergoing direct impacts.
+    years : np.array
+        Years of the considered hazard events for which impact is calculated.
+    direct_impact : np.array
+        Direct impact array.
+    direct_aai_agg : np.array
+        Average annual direct impact array.
+    indirect_impact : np.array
+        Indirect impact array.
+    indirect_aai_agg : np.array
+        Average annual indirect impact array.
+    total_impact : np.array
+        Total impact array.
+    total_aai_agg : np.array
+        Average annual total impact array.
+    io_data : dict
+        Dictionary with the coefficients, inverse and risk_structure matrixes and
+        the selected input-output modeling approach.
+    """
+
+    def __init__(self):
+        """Initialize SupplyChain."""
+        self.mriot_data = np.array([], dtype='f')
+        self.mriot_reg_names = np.array([], dtype='str')
+        self.sectors = np.array([], dtype='str')
+        self.total_prod = np.array([], dtype='f')
+        self.mriot_type = ''
+        self.reg_pos = {}
+        self.years = np.array([], dtype='f')
+        self.direct_impact = np.array([], dtype='f')
+        self.direct_aai_agg = np.array([], dtype='f')
+        self.indirect_impact = np.array([], dtype='f')
+        self.indirect_aai_agg = np.array([], dtype='f')
+        self.total_impact = np.array([], dtype='f')
+        self.total_aai_agg = np.array([], dtype='f')
+        self.io_data = {}
+
+    def read_wiod16(self, year=2014,
+                    table_info_pos={'range_rows':(5, 2469),
+                                    'range_cols':(4, 2468),
+                                    'col_iso3':2,'col_sectors':1}):
```

Comment:
```suggestion
                    range_rows=(5, 2469),
                    range_cols=(4, 2468),
                    col_iso3=2, col_sectors=1):
```
this would imply changes in the code and docstring, of course.

Reply:
Another possibility is 
```
(...
table_info_post = None):
if table_info_post is None:
   table_info_pos={'range_rows':(5, 2469),
                              'range_cols':(4, 2468),
                               'col_iso3':2,
                               'col_sectors':1}
```